protocol/error.go

```diff
@@ -15,13 +15,33 @@ const (
 	ErrorMalformedClientMessage
 )
 
+const (
+	Passed ErrorCode = iota + 20
```

Comment:
Do you mean to try to map HTTP status codes to our errors? Or just the idea of assigning say errors in the 500s to server errors? This is similar to what we're already doing, but would give us a larger error space.